packages/dashboard/src/helpers/updateChildFilters.ts

```diff
@@ -0,0 +1,48 @@
+import { SharedFilter } from '../types/SharedFilter'
+import _ from 'lodash'
+
+export const updateChildFilters = (newSharedFilters: SharedFilter[], data: Record<string, any>): SharedFilter[] => {
+  const dataSet = Object.values(data).flat()
+  // filter out Data type filters
+  const dataFilters = newSharedFilters.filter(filter => filter.type === 'datafilter')
+  const urlFilters = newSharedFilters.filter(filter => filter.type === 'urlfilter')
+  // Find indexes of all child filters
+  const childFilterIndexes: number[] = dataFilters
+    .map((filter, index) => (filter.parents ? index : -1))
+    .filter(index => index !== -1)
+  if (childFilterIndexes.length === 0) return newSharedFilters
+
+  // deep copy of the shared filters
+  const updatedFilters = _.cloneDeep(dataFilters)
+
+  // Update each child filter
+  childFilterIndexes.forEach(childIndex => {
+    const childFilter: SharedFilter = dataFilters[childIndex]
+    const parentFilter: SharedFilter = dataFilters.find(filter => String(childFilter.parents) === String(filter.key))
+    const isParentMultiSelect = parentFilter.filterStyle === 'multi-select'
+
+    if (parentFilter) {
+      // Filter dataset based on parent's active value
+      const parentsActiveValues: string[] = dataSet.filter((d: Record<string, any>) => {
+        if (isParentMultiSelect) {
+          // If multi-select, check if the parent active array includes the value
+          return Array.isArray(parentFilter.active) && parentFilter.active.includes(d[parentFilter.columnName])
+        } else {
+          return parentFilter.active === d[parentFilter.columnName]
+        }
+      })
+      // Get unique active values for the child filter
+      const uniqChildValues = _.uniq(parentsActiveValues.map(d => d[childFilter.columnName]).filter(Boolean))
+
+      // Update the child filter if unique values exist
+      if (uniqChildValues.length > 0) {
+        updatedFilters[childIndex] = {
+          ...childFilter,
+          values: uniqChildValues
+        }
+      }
+    }
+  })
+
+  return [...updatedFilters, ...urlFilters]
```

Comment:
Won't this reorder sharedFilter indexes? They should stay in the same place as the dashboardFilter visualizations are referencing those indexes.

Reply:
@joshlacey  moat likely will re order and create mess. i will find solution for next sprint.

Reply:
I think we need to find a solution before committing this code. Can we push this ticket to next sprint?

Reply:
@Atash3000 what is the status here?

Reply:
@adamdoe  waitings for Josh to check it.